server.js

```diff
@@ -7967,6 +7967,86 @@ cache(function (data, match, sendBadge, request) {
   });
 }));
 
+// Launchpad integration
+camp.route(/^\/launchpad\/ppa\/([^/]+)\/([^/]+)\/([^/]+)\/([^/]+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  // The type of badge, either version or downloads
+  var type = match[1];
+  // The owner of the ppa
+  var user = match[2];
+  // The name of the ppa
+  var ppa = match[3];
+  // The package in the PPA to use
+  var name = match[4];
+  var format = match[5];
+
+  var badgeData;
+
+  if (type == 'version') {
+    badgeData = getBadgeData('version', 'invalid');
+  }
+  else if (type == 'downloads') {
+    badgeData = getBadgeData('downloads 24h', 'invalid');
+  }
+
+  // Make a request for packages in the PPA
+  var apiUrl = 'https://api.launchpad.net/1.0/~' + user + '/+archive/' + ppa + '?ws.op=getPublishedBinaries&ws.size=150';
+  request(apiUrl, function(err, res, buffer) {
```

Comment:
https://github.com/badges/shields/blob/2ded83f8a68cd36102167883d0f9008a7404cb24/lib/error-helper.js#L8
is a great way to check response before parsing response body.